### PR TITLE
[Refactor] Integrated some lightllm kernels into token-attention 

### PR DESCRIPTION
**Pull request by tiandiao123**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [ ] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?
This PR is used integrate some triton kernels from lightllm into our inference system. 

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [ ] I have performed a self-review of my code
- [ ] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [ ] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: colossalai/inference/tensor_parallel/kvcache_manager.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
-# Adapted from lightllm/common/mem_manager.py
+# Refered/Modified from lightllm/common/mem_manager.py
 # of the ModelTC/lightllm GitHub repository
```
Comment (CjhHa1) [inline colossalai/inference/tensor_parallel/kvcache_manager.py:2]: consider use '""" """"'

File: colossalai/inference/tensor_parallel/modeling/chatglm2.py [hunk 2]
```diff
@@ -20,6 +18,15 @@
 
 from ._utils import copy_kv_to_mem_cache
 
+try:
+    from lightllm.models.llama2.triton_kernel.context_flashattention_nopad import context_attention_fwd as lightllm_llama2_context_attention_fwd
+    from lightllm.models.chatglm2.triton_kernel.rotary_emb import rotary_emb_fwd as chatglm2_rotary_emb_fwd
+    print("found lightllm installation for inference")
+    HAS_LIGHTLLM_KERNEL = True
+except:
+    print("please install lightllm from source to run inference: ")
```
Comment (CjhHa1) [inline colossalai/inference/tensor_parallel/modeling/chatglm2.py:27]: add a link after ':'?

File: colossalai/inference/tensor_parallel/batch_infer_state.py [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
 from .kvcache_manager import MemoryManager
 
-
+# adaped from: lightllm/server/router/model_infer/infer_batch.py
```
Comment (Xu-Kai) [inline colossalai/inference/tensor_parallel/batch_infer_state.py:8]: Adaped should be adapted. Follows are same error.

Comment (tiandiao123) [reply colossalai/inference/tensor_parallel/batch_infer_state.py:8]: fixed .

File: colossalai/inference/tensor_parallel/modeling/chatglm2.py (hunk 2 above)
Comment (tiandiao123) [reply colossalai/inference/tensor_parallel/modeling/chatglm2.py:27]: fixed .

File: colossalai/inference/tensor_parallel/kvcache_manager.py (hunk 1 above)
Comment (tiandiao123) [reply colossalai/inference/tensor_parallel/kvcache_manager.py:2]: ok !

File: examples/inference/bench_llama.py [hunk 4]
```diff
@@ -13,7 +13,7 @@
 from colossalai.testing import clear_cache_before_run, rerun_if_address_is_in_use, spawn
 
 os.environ["TRANSFORMERS_NO_ADVISORY_WARNINGS"] = "true"
-
+torch.cuda.empty_cache()
```
Comment (yuanheng-zhao) [inline examples/inference/bench_llama.py:16]: Better to move it into a function.

Comment (tiandiao123) [reply examples/inference/bench_llama.py:16]: OK!

File: colossalai/inference/tensor_parallel/modeling/llama.py [hunk 5]
```diff
@@ -29,6 +26,15 @@
     )
     HAS_VLLM_KERNERL = False
 
+try:
+    from lightllm.models.llama2.triton_kernel.context_flashattention_nopad import context_attention_fwd as lightllm_llama2_context_attention_fwd
+    from lightllm.models.llama.triton_kernel.rotary_emb import rotary_emb_fwd as llama_rotary_embedding_fwd
+    print("found lightllm installation for inference")
```
[the remark this answers is not in the document]
Comment (CjhHa1) [reply colossalai/inference/tensor_parallel/modeling/llama.py]: done !

**Review by CjhHa1 — APPROVED** [review]
LGTM

**Pull request merged**